src/build/index.js

```diff
@@ -115,5 +123,5 @@ export default function build(source, config = {}, typeDeps = [], infer = true)
   }
 
   // return a type map
-  return types.reduce((map, type) => ({ ...map, [type.name]: type }), {});
+  return types.reduce((map, type) => ({ ...map, [type.name]: type }), { });
```

Comment:
It seems to me that what lines 96 - 106 are doing could be changed to:

```js
return buildSchema(
  schemaASTs[0],
  config[SCHEMA_CONFIG_KEY],
  () => ensureNoDuplicateTypes([...types, ...typeDeps]),
);
```

The union of the type dependencies and schema types into `schemaConfig.types` could cause unexpected side effects as the schema's types and the type dependencies for the template string are 2 different things.